ui/app/src/views/auth/SignWrapper.tsx

```diff
@@ -11,26 +11,51 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { Button, Divider, Stack } from '@mui/material';
+import { Divider, Stack } from '@mui/material';
 import { ReactNode } from 'react';
+import {
+  BufferLoginButton,
+  GithubLoginButton,
+  GoogleLoginButton,
+  MicrosoftLoginButton,
+} from 'react-social-login-buttons';
+import * as React from 'react';
 import { useDarkMode } from '../../context/DarkMode';
 import PersesLogoCropped from '../../components/logo/PersesLogoCropped';
 import DarkThemePersesLogo from '../../components/logo/DarkThemePersesLogo';
 import LightThemePersesLogo from '../../components/logo/LightThemePersesLogo';
 import { useIsLaptopSize } from '../../utils/browser-size';
 import { useConfigContext } from '../../context/Config';
 
+function computeSocialButtonFromURL(url: string) {
+  if (url.includes('microsoft')) {
+    return MicrosoftLoginButton;
+  }
+  if (url.includes('github')) {
+    return GithubLoginButton;
+  }
+  if (url.includes('google')) {
+    return GoogleLoginButton;
+  }
+  //TODO: Support more providers managed by the lib
+  //TODO: Support more providers not managed by the lib (e.g Gitlab)
```

Comment:
we could try to propose a new button to the upstream lib. WDYT @celian-garcia ?

Reply:
Yes indeed that'd probably be well approved.

Reply:
PR proposed for Gitlab https://github.com/michaeljscript/react-social-login-buttons/pull/115